designs/2024-support-l10n-rule-messages/README.md

```diff
@@ -0,0 +1,185 @@
+   This way should allow users to provide some plugins to update other plugins' messages. This approach allows users to customize the messages for rules provided by plugins.
+
+   Example:
+    
+   ```js
+   module.exports = {
+     locale: "es",
+     rules: [/* ... */],
+     messages: {
+       "plugin/ruleId": {
+         someMessageId: "Mensaje predeterminado en español"
+       }
+     }
+   };
+   ```
+
+   and eslint will use the message info to replace the message in the rule if the message ID is found in the configuration.
+
+for corner cases, we need to consider the following:
+
+1. If a translation is missing for a specific language, ESLint should fall back to the default English message.
+2. ESLint should support original flat rules structure and the new structure with localized messages.
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+1. Update the official ESLint documentation to include information on how to create and use localization files.
+2. Provide examples and guidelines for contributing translations, including third party integrations.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think 
+    about any opposing viewpoints that reviewers might bring up. 
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+1. Increased Maintenance Burden: Adding support for localization will increase the complexity of the ESLint codebase. Maintaining translations and ensuring they are up-to-date with rule changes will require additional effort from both the core team and contributors.
+2. It's better to use a module to handle the localization such as typescript using `@types/some-module` but it's not possible to use a module to handle the localization in ESLint.
+3. Potential for Incomplete Translations: There is a risk that some languages may have incomplete translations, leading to a mix of localized and default English messages.
+   This could confuse users and reduce the overall effectiveness of the feature.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+This change should not affect existing ESLint users who do not use the localization feature. Users who do not provide translations will continue to see the default English messages. To minimize disruption, ESLint should provide clear documentation on how to create and use localization files, as well as guidelines for contributing translations.
```

Comment:
Backwards compatibility analysis should consider plugin compatibility as well as end user / config compatibility.  What's the implication to plugin authors who haven't set up the locale-based message ids?